doc/dev/adr/0023-property-support.md

```diff
@@ -0,0 +1,312 @@
+
+* For statically itemized properties, the optional fields in the `TProp` response object contain values for any Properties that are in the notify list.
+* For dynamically itemized properties, the key/value pairs in the `TProp` response map indicate which Properties are in the notify list and their corresponding values.
+
+## Additive change to DTDL Mqtt extension
+
+To support MQTT communication of DTDL Property contents, the Mqtt extension will be updated.
+Specifically, when the Mqtt adjunct type co-types an Interface, it adds properties `commandTopic`, `telemetryTopic`, and [several others](https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v4/DTDL.mqtt.v3.md#mqtt-adjunct-type).
+As part of the present change, an additional property will be added:
+
+| Property | Required | Data type | Limits | Description |
+| --- | --- | --- | --- | --- |
+| `propertyTopic` | optional | *string* | slash-separated sequence of character-restricted labels and/or brace-enclosed tokens | MQTT topic pattern on which a request or notification is published. |
+
+The DTDL Mqtt extension places no restrictions &mdash; other than basic syntactical constraints &mdash; on the set of tokens used in MQTT topic patterns.
+Topic tokens recognized by the ProtocolCompiler are defined in the [topic-structure](https://github.com/Azure/iot-operations-sdks/blob/main/doc/reference/topic-structure.md) document.
+The sets of tokens differ between RPC and Telemetry, and the following set of tokens is hereby defined for Property:
+
+| Topic token | Description | Required |
+| --- | --- | --- |
+| `{modelId}` | The identifier of the service model, which is the full DTMI of the Interface, might include the version | optional |
+| `{maintainerId}` | Identifier of the maintainer, by default the MQTT client ID | optional |
+| `{sourceId}` | Identifier of the source of the request or notification, by default the MQTT client ID | optional |
+| `{action}` | One of "read", "write", "observe", "unobserve", or "notify" | optional |
+
+An example Property topic pattern is illustrated in the sample model below.
+
+## Sample model
+
+The following DTDL model defines three Properties, two of which ("Foo" and "Bar") are statically itemized, and one of which ("Props") is dynamically itemized.
+These Properties correspond to the example types [defined above](#property-envoy).
+
+```json
+{
+  "@context": [ "dtmi:dtdl:context;4", "dtmi:dtdl:extension:mqtt;4" ],
+  "@id": "dtmi:propertySketch:PropertySketch;1",
+  "@type": [ "Interface", "Mqtt" ],
+  "payloadFormat": "Json/ecma/404",
+  "propertyTopic": "sample/{modelId}/{sourceId}/property/{action}",
+  "contents": [
+    {
+      "@type": "Property",
+      "name": "Foo",
+      "schema": "integer"
+    },
+    {
+      "@type": "Property",
+      "name": "Bar",
+      "schema": "string"
+    },
+    {
+      "@type": [ "Property", "Fragmented" ],
```

Comment:
Should dynamically itemized properties be defined in the same chunk of DTDL as the statically itemized properties since they won't be generated together (but the statically itemized properties would be)?

Reply:
In fact, static and dynamic Properties *should* be generated together.  This was a weakness of the ADR that I have fixed with the latest commit.